src/DefaultCachedRepository.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace BasicRepos;
+
+public class DefaultCachedRepository<TEntity, TContext> : RepositoryBase<TEntity>, ICachedRepository<TEntity>
+where TEntity : class
+where TContext : DbContext
+{
+    protected IEnumerable<TEntity> cachedEntities { get; set; } = new List<TEntity>();
+
+    public DefaultCachedRepository(TContext dbFactory) : base(dbFactory)
+    {
+    }
+
+    public override async Task<IEnumerable<TEntity>> GetAllAsync(Expression<Func<TEntity, bool>> predicate, CancellationToken cancellationToken = default)
```

Comment:
Should this pass the cancellation token to the cache refresh so that if cancellation is issued it properly cascades into the await chain that feeds into the getallasync?

Reply:
Ah yes. This also reminds me that we should probably expose the RefreshCache as public